From 0b5ba1e7a39b7801f677fc2b1ec8971ba53e3154 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 21 Feb 2013 16:51:00 +0800
Subject: [PATCH 162/602] ENGR00236850 WFD: add audio remote submix module

1.The remote submix module "audio.r_submix.default" is a must for WFD.
  For we have no change for it, so still use default not imx*.
2.Add path config in audio policy config. Align all boards expect
 sabreauto. No requirement for auto I think.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 evk_6sl/audio_policy.conf        | 28 ++++++++++++++++++++++++++++
 hdmidongle_6dq/audio_policy.conf | 28 ++++++++++++++++++++++++++++
 imx53_smd/audio_policy.conf      | 18 ++++++++++++++++++
 imx5x/imx5x.mk                   |  1 +
 imx6/imx6.mk                     |  1 +
 sabresd_6dq/audio_policy.conf    | 28 ++++++++++++++++++++++++++++
 6 files changed, 104 insertions(+)

diff --git a/evk_6sl/audio_policy.conf b/evk_6sl/audio_policy.conf
index 59573db..2fca29c 100644
--- a/evk_6sl/audio_policy.conf
+++ b/evk_6sl/audio_policy.conf
@@ -46,4 +46,32 @@ audio_hw_modules {
       }
     }
   }
+  a2dp {
+    outputs {
+      a2dp {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_ALL_A2DP
+      }
+    }
+  }
+  r_submix {
+    outputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+      }
+    }
+    inputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
+      }
+    }
+  }
 }
diff --git a/hdmidongle_6dq/audio_policy.conf b/hdmidongle_6dq/audio_policy.conf
index 59573db..2fca29c 100644
--- a/hdmidongle_6dq/audio_policy.conf
+++ b/hdmidongle_6dq/audio_policy.conf
@@ -46,4 +46,32 @@ audio_hw_modules {
       }
     }
   }
+  a2dp {
+    outputs {
+      a2dp {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_ALL_A2DP
+      }
+    }
+  }
+  r_submix {
+    outputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+      }
+    }
+    inputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
+      }
+    }
+  }
 }
diff --git a/imx53_smd/audio_policy.conf b/imx53_smd/audio_policy.conf
index 6ec67b1..606e61f 100644
--- a/imx53_smd/audio_policy.conf
+++ b/imx53_smd/audio_policy.conf
@@ -72,4 +72,22 @@ audio_hw_modules {
       }
     }
   }
+  r_submix {
+    outputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+      }
+    }
+    inputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
+      }
+    }
+  }
 }
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index e8f5fd5..2a052c2 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -76,6 +76,7 @@ PRODUCT_PACKAGES += \
 	power.imx5x				\
 	libbt-vendor				\
 	bt_vendor.conf				\
+        audio.r_submix.default                  \
 	magd
 
 # Bluetooth firmware files.
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index c7a1397..811c55c 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -83,6 +83,7 @@ PRODUCT_PACKAGES += \
 	hwcomposer.imx6             		\
 	camera.imx6                 		\
 	power.imx6				\
+        audio.r_submix.default                  \
 	magd
 
 # Bluetooth firmware files.
diff --git a/sabresd_6dq/audio_policy.conf b/sabresd_6dq/audio_policy.conf
index 59573db..2fca29c 100644
--- a/sabresd_6dq/audio_policy.conf
+++ b/sabresd_6dq/audio_policy.conf
@@ -46,4 +46,32 @@ audio_hw_modules {
       }
     }
   }
+  a2dp {
+    outputs {
+      a2dp {
+        sampling_rates 44100
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_ALL_A2DP
+      }
+    }
+  }
+  r_submix {
+    outputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_OUT_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_OUT_REMOTE_SUBMIX
+      }
+    }
+    inputs {
+      submix {
+        sampling_rates 44100|48000
+        channel_masks AUDIO_CHANNEL_IN_STEREO
+        formats AUDIO_FORMAT_PCM_16_BIT
+        devices AUDIO_DEVICE_IN_REMOTE_SUBMIX
+      }
+    }
+  }
 }
-- 
1.8.0

